sdks/python/apache_beam/transforms/window.py

```diff
@@ -245,10 +274,17 @@ def __init__(self, value, timestamp):
     self.value = value
     self.timestamp = Timestamp.of(timestamp)
 
-  def __cmp__(self, other):
+  def __eq__(self, other):
+    return (type(self) == type(other)) and (self.value == other.value) and \
+           (self.timestamp == other.timestamp)
+
+  def __hash__(self):
+    return hash((type(self), self.value, self.timestamp))
+
+  def __lt__(self, other):
     if type(self) is not type(other):
-      return cmp(type(self), type(other))
-    return cmp((self.value, self.timestamp), (other.value, other.timestamp))
+      return type(self) < type(other)
```

Comment:
I fear this might fail since types are not comparable in Python 3 unless you use a custom metaclass (which is not the case here).

Reply:
Indeed! Using the hash of type should resolve this issue.